src/react/databrowser/objects/__tests__/Metadata.test.jsx

```diff
@@ -0,0 +1,137 @@
+
+        // click on add button
+        component.find('Button#addbtn0').simulate('click');
+
+        // check if new item row added
+        expect(component.find(Item)).toHaveLength(2);
+
+        // check if function when pressing save button is triggered
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(0);
+        component.find('Button#metadata-button-save').simulate('click');
+        expect(putObjectMetadataMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('remove button and add button should be disabled', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
+
+        // check if item row values are rendered
+        expect(component.find('Button#addbtn0').prop('disabled')).toBe(true);
+        expect(component.find('Button#delbtn0').prop('disabled')).toBe(true);
+    });
+
+    it('should render SelectBox with key/value pass in props', () => {
+        const { component } = reduxMount(<Metadata
+            objectMetadata={{ ...OBJECT_METADATA, metadata: [{ key: 'CacheControl', value: 'no-cache', type: METADATA_SYSTEM_TYPE }] }}/>);
+
+        // check if only one item row is rendered
+        const items = component.find(Item);
+        expect(items).toHaveLength(1);
+
+        // check if item row values are rendered
+        const firstItem = items.first();
+        expect(firstItem.find('input[name="mdKeyType"]').prop('value')).toBe('CacheControl');
+        expect(firstItem.find('input#metadata-input-value').prop('value')).toBe('no-cache');
+    });
+
+    it('should delete key/value if remove button is pressed', () => {
+        const { component } = reduxMount(<Metadata objectMetadata={{
+            ...OBJECT_METADATA,
+            metadata: [{ key: 'CacheControl', value: 'no-cache', type: METADATA_SYSTEM_TYPE }, { key: 'cache-type', value: '1', type: METADATA_USER_TYPE }],
+        }}/>);
+
+        // check if two item rows are rendered
+        let items = component.find(Item);
+        expect(items).toHaveLength(2);
+
+        // check if first item row values are rendered
+        let firstItem = items.first();
+        expect(firstItem.find('input[name="mdKeyType"]').prop('value')).toBe('CacheControl');
+        expect(firstItem.find('input#metadata-input-value').prop('value')).toBe('no-cache');
+
+        // check if second item row values are rendered
+        let secondItem = items.at(1);
+        expect(secondItem.find('input[name="mdKeyType"]').prop('value')).toBe('x-amz-meta');
+        expect(secondItem.find('input#metadata-input-extra-key').prop('value')).toBe('cache-type');
+        expect(secondItem.find('input#metadata-input-value').prop('value')).toBe('1');
+
+        // check if first AddButton is hidden and second one is visible
+        expect(component.find('Button#addbtn0').prop('isVisible')).toBe(false);
```

Comment:
We can use `firstItem` and `secondItem` instead of `component` to make sure we are in the first `Item`. 
Same thing for the following assertions.

Reply:
done